src/init/raw_data.rs

```diff
@@ -31,6 +31,7 @@ pub async fn grab_raw_data(input: &Input) -> Result<RawDataResults> {
     let mut osm_needed = BTreeSet::new();
     for rec in csv::Reader::from_reader(File::open(lookup_path)?).deserialize() {
         let rec: LookupRow = rec?;
+        results.oa_to_msoa.insert(rec.oa, rec.msoa.clone());
```

Comment:
Making sure I understand the fix: the special areas wound up including OAs that were outside of the MSOAs specified in the config files?

Reply:
Yeah that's it - additional OAs from the population files that are loaded conditional on the input MSOAs need to be looked up.

Here's my fuller understanding (in case helpful and for future reference): the population files (the "azure_ref" ones) to be loaded are determined in [raw_data](https://github.com/alan-turing-institute/uatk-spc/blob/f0f1f8cf14fca225377d2b7c29bc5267871a2b4e/src/init/raw_data.rs#L34-L42) based on the input MSOAs. Currently only OAs in those input MSOAs get included in the lookup (as they are [inside the "if"](https://github.com/alan-turing-institute/uatk-spc/blob/f0f1f8cf14fca225377d2b7c29bc5267871a2b4e/src/init/raw_data.rs#L42)). When reading those population files, the MSOA of each individual is looked up using [`oa_to_msoa`](https://github.com/alan-turing-institute/uatk-spc/blob/f0f1f8cf14fca225377d2b7c29bc5267871a2b4e/src/init/population.rs#L42-L45). For non-special areas, all the MSOAs for a given population file are included in the input so the `oa_to_msoa` lookup has all the encountered OAs. However, for the special areas they do not always contain all the MSOAs of the population files so we currently encounter some OAs [here](https://github.com/alan-turing-institute/uatk-spc/blob/f0f1f8cf14fca225377d2b7c29bc5267871a2b4e/src/init/raw_data.rs#L42) that are not in the lookup.

Populating the `oa_to_msoa` lookup for all OAs outside the "if" resolves this as all OAs are then in the lookup. And then we still disregard MSOAs not included in the input [here](https://github.com/alan-turing-institute/uatk-spc/blob/f0f1f8cf14fca225377d2b7c29bc5267871a2b4e/src/init/population.rs#L49).